### PR TITLE
Persist and sync

### PR DESCRIPTION
**Pull request by mayank1513**
## Related Issues or Discussions

Fixes #

## Summary

Added third party integration

- [persist-and-sync](https://www.npmjs.com/package/persist-and-sync) - Zustand middleware to easily persist and sync Zustand state between tabs / windows / iframes (Same Origin)

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,6 @@ package-lock.json
 coverage/
 .rpt2_cache/
 .idea
+
+# lock files
+*lock*
```
Comment (sewera) [inline .gitignore:16]: `yarn.lock` file is and should remain tracked by Git

File: docs/integrations/third-party-libraries.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This can be done using third-party libraries created by the community.
 - [leiten-zustand](https://github.com/hecmatyar/leiten-zustand) — Cleans your store from boilerplate for requests and data transformation.
 - [mobz](https://github.com/2A5F/Mobz) — Zustand-style MobX API.
 - [ngx-zustand](https://github.com/JoaoPauloLousada/ngx-zustand) - A Zustand adapter for Angular.
+- [persist-and-sync](https://www.npmjs.com/package/persist-and-sync) - Zustand middleware to easily persist and sync Zustand state between tabs / windows / iframes (SameOrigin)
```
Comment (sewera) [inline docs/integrations/third-party-libraries.md:26]: Do you have a Github link for it?

**Review by sewera — CHANGES_REQUESTED** [review]

File: docs/integrations/third-party-libraries.md (hunk 2 above)
Comment (mayank1513) [reply docs/integrations/third-party-libraries.md]: Added GitHub link and also updated npm

**Comment by dai-shi** [comment]
Please don't modify .gitignore in this PR.

**Comment by mayank1513** [comment]
> Please don't modify .gitignore in this PR.

ok. So, you want me to create separate PR for .gitignore?

**Comment by dai-shi** [comment]
Sorry for the confusion. At this point, I don't think we need to modify .gitignore. If you want to discuss, you may want to open a new discussion first.

File: docs/integrations/third-party-libraries.md [hunk 3]
```diff
@@ -23,6 +23,7 @@ This can be done using third-party libraries created by the community.
 - [leiten-zustand](https://github.com/hecmatyar/leiten-zustand) — Cleans your store from boilerplate for requests and data transformation.
 - [mobz](https://github.com/2A5F/Mobz) — Zustand-style MobX API.
 - [ngx-zustand](https://github.com/JoaoPauloLousada/ngx-zustand) - A Zustand adapter for Angular.
+- [persist-and-sync](https://github.com/mayank1513/persist-and-sync) - Zustand middleware to easily persist and sync Zustand state between tabs / windows / iframes (SameOrigin) ![npm bundle size](https://img.shields.io/bundlephobia/minzip/persist-and-sync)
```
Comment (dai-shi) [inline docs/integrations/third-party-libraries.md:26]: Let's make it consistent with others.
```suggestion
- [persist-and-sync](https://github.com/mayank1513/persist-and-sync) - Zustand middleware to easily persist and sync Zustand state between tabs / windows / iframes (SameOrigin)
```

File: docs/integrations/third-party-libraries.md [hunk 4]
```diff
@@ -23,6 +23,7 @@ This can be done using third-party libraries created by the community.
 - [leiten-zustand](https://github.com/hecmatyar/leiten-zustand) — Cleans your store from boilerplate for requests and data transformation.
 - [mobz](https://github.com/2A5F/Mobz) — Zustand-style MobX API.
 - [ngx-zustand](https://github.com/JoaoPauloLousada/ngx-zustand) - A Zustand adapter for Angular.
+- [persist-and-sync](https://github.com/mayank1513/persist-and-sync) - Zustand middleware to easily persist and sync Zustand state between tabs / windows / iframes (SameOrigin)
```
Comment (sewera) [inline docs/integrations/third-party-libraries.md:26]: You can add a full stop at the end and rephrase "SameOrigin" to something like "for apps from the same origin."

**Review by sewera — CHANGES_REQUESTED** [review]
You can also add this library to the previous PR so that your changes are all in one place if you want.

**Review by sewera — APPROVED** [review]

**Pull request merged**